### PR TITLE
Beyza GOK WEEK3

### PR DESCRIPTION
**Pull request by beyzagoknl**

### CODE REVIEW

**Comment by beyzagoknl** [comment]
https://zesty-raindrop-2d9fae.netlify.app/

File: week3/project/ecommerce/src/components/ProductsItem.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { ReactComponent as SolidHeart } from "../assets/heart-solid.svg";
+
+function ProductsItem({ item }) {
+  const [favorite, handleFavorite] = useContext(FavoriteContext);
+  const favoriteItem = favorite.find((favoriteId) => favoriteId === item.id);
+
+  return (
+    <li className="products-item">
+      <div className="product">
+        <div className="heart-image" onClick={() => handleFavorite(item.id)}>
```
Comment (thibaudszy) [inline week3/project/ecommerce/src/components/ProductsItem.js:14]: A button element would be more appropriate here. Avoid clickable divs when you can

File: week3/project/ecommerce/src/components/ProductDetails.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  if (loading) return <Loading />;
+  if (error) return <Error text="Not load the products." />;
+
+  return (
+    <div className="App">
+      <Header title={product.title} />
+
+      <div className="product-details">
+        <div className="image-container">
+          <div className="heart-image" onClick={() => handleFavorite(+id)}>
```
Comment (thibaudszy) [inline week3/project/ecommerce/src/components/ProductDetails.js:30]: Why the `+` before the id?

File: week3/project/ecommerce/src/components/Products.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import Loading from "./Loading";
+import Error from "./Error";
+
+const Products = ({ category }) => {
+  const productsUrl = `https://fakestoreapi.com/products`;
+  const categoryUrl = `https://fakestoreapi.com/products/category/${category}`;
+  const url = category === "allCategories" ? productsUrl : categoryUrl;
+
+  const [products, loading, error] = useFetchData(url);
+  if (loading) return <Loading />;
```
Comment (thibaudszy) [inline week3/project/ecommerce/src/components/Products.js:12]: I like this format! It's easier to read than chaining ternary operators.

**Review by thibaudszy — COMMENTED** [review]
Nice work. 
Putting the function to fetch multiple items in the `useFetchData` hook is a nice idea but in the future, I'd suggest putting it in a separate hook. It's better to keep your hooks as simple as possible. 

I have a remark about your file structure: it is recommended to have a `pages` directory at the same level as the `components` one containing your views (the components loaded in  `app.js`). This makes the code easier to navigate.

**Pull request closed without merging**